OpenRA.Mods.Common/Traits/AutoTarget.cs

```diff
@@ -68,6 +69,7 @@ public AutoTarget(ActorInitializer init, AutoTargetInfo info)
 		{
 			var self = init.Self;
 			attackBases = self.TraitsImplementing<AttackBase>().ToArray();
+			activeAttackBases = attackBases.Where(Exts.IsTraitEnabled);
```

Comment:
Do you think it's worth changing line 110 to `!activeAttackBases.Any(a => a.HasAnyValidWeapons(Target.FromActor(attacker))` (that repeated `FromActor` also looks wasteful) and then removing the `attackBases` member?